README.md

```diff
@@ -154,37 +154,105 @@ See [Extensions](#extensions) for captioning-specific command line options. Chec
-images of the validation dataset. 
+Inference can be performed using the `inference.py` script to generate captions for images of a specified dataset.    
+The model parameters, a file containing image IDs and the dataset-split these image IDs are taken from have to be supplied.
+An optional output path may be specified to store the predicted captions as a json-file.
+
+The following command shows an example usage of the script:
 
 ```
-python demo.py \
+python inference.py \
        --features grid \
        --features-dir output \
        --captions-dir output \
        --user-dir task \
        --tokenizer moses \
        --bpe subword_nmt \
-        --bpe-codes output/codes.txt \
+       --bpe-codes output/codes.txt \
        --beam 5 \
        --path .checkpoints/checkpoint_best.pt \
-       --input demo/val-images.txt
+       --split valid \
+       --input demo/val-images.txt \
+       --output demo/val-predictions.json
 ```
 
-
-
-Validation image IDs are read from [demo/val-images.txt](demo/val-images.txt). This should produce an output containing 
-something like 
+Image IDs are read from [demo/val-images.txt](demo/val-images.txt) in this sample. 
+This should produce an output containing something like:
 
 ```
 105537: A street sign hanging from the side of a metal pole.
 130599: A man standing next to a giraffe statue.
 ...
 ```
 
+Furthermore, the predictions are stored in [demo/val-predictions.json](demo/val-predictions.json).
+
+## Evaluation
+
+### Installation of dependencies
+
+Evaluation is performed using a [Python 3 Fork](https://github.com/flauted/coco-caption) of the
+[COCO Caption Evaluation](https://github.com/tylin/coco-caption) library. This library is included
+as a [Git Submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) in the path `external/coco-caption`.
+
+Download the submodule using following commands:
+
+```
+git submodule init
+git submodule update
+``` 
+
+Furthermore, the COCO Caption Evaluation library uses the [Stanford CoreNLP 3.6.0](https://stanfordnlp.github.io/CoreNLP/index.html) toolset
+which must be downloaded prior to the first execution of the scoring script. 
+
+It is important to change into the submodules root folder before executing the script to download the required files:
+
+```
+cd external/coco-caption
+./get_stanford_models.sh
+```
+
+### Model evaluation
+
+Evaluation of a model is performed using the `score.py` script.
+This script uses the specified reference-captions as the ground truth and evaluates the model based on the generated captions provided as a json-file (created prior via the `inference.py` script).    
+    
+The following example calculates the evaluation scores for the generated captions provided in [demo/val-predictions.json](demo/val-predictions.json).
+
+```
+python score.py \
+       --reference-captions external/coco-caption/annotations/captions_val2014.json \
+       --system-captions demo/val-predictions.json
+```
```

Comment:
I first had to verify for myself that the Karpathy `valid` split used above is actually a subset of the COCO `val2014` split. Maybe it is helpful to add a corresponding note.